### ISSUE TITLE
Add an Option to Allow Auto Continuing Playback from Last Position

### ISSUE THREAD

**Issue by asankaSovis** [Rafostar/clapper#442]
You should resume playback from the last point for videos when opened. This is available in most if not all media players and from my understanding, Clapper also had an option for this at some point. This would be a good option to have for the convenience of users.

**Comment by Rafostar** [comment]
Yes, it was a bad implementation, so I decided to not port that as-is into new code. It needs to be rethought carefully how it will suit and work well with current UI.

**Comment by 64-bitman** [comment]
Any progress on this?

**Comment by j-a-sunny** [comment]
still nothing?

**Comment by Rafostar** [comment]
> still nothing?

Lots of issues, one me. If you want something done faster, contributions are welcome. Still, not entirely "nothing" as we now have initial plugin system since 0.8 version. I am hoping this can be written as one of these plugins (to avoid increasing dependencies) once we have a good base interface for implementing such functionality.

**Comment by abhijeeth-babu** [comment]
> > still nothing?
> 
> Lots of issues, one me. If you want something done faster, contributions are welcome. Still, not entirely "nothing" as we now have initial plugin system since 0.8 version. I am hoping this can be written as one of these plugins (to avoid increasing dependencies) once we have a good base interface for implementing such functionality.

Hi, is there any documentation available on how to implement plugins for clapper using libpeas?

**Comment by Rafostar** [comment]
> I need to design and implement an interface for writing plugins that can read current media item and have access to the player first.

_Progress update:_ There is now such interface, so no more blockers on this anymore I guess 😄

**Comment by asankaSovis** [comment]
@Rafostar Appreciate your effort. Waiting to see it in the next update.

**Comment by Rafostar** [comment]
I implemented this as an enhancer plugin. Once installed (Flatpak will include it) you can tweak/configure it in app preferences window. I hope that everyone will find the combination that suits them the best.

For example you can set minimal amount of time watched and/or until the end for position to be remembered. You can also set minimal video duration, so you can make it work only for longer videos like movies. Lastly, whether you wish for it to automatically resume playback, mark the resume point on the position/seek bar or do both.

Marking position with indicator on seek bar is also a replacement of previous "ask to resume" functionality. I concluded that a popup with a question whether to resume is useless and feels like an advert. So just marking the point on seek bar and its still 1 click to resume when you click on it and zero clicks if you do not want to resume (you just ignore it).

For local files I put an extra effort to make it still work even when video file is renamed and/or moved to a different directory (not every player handles this with resume functionality).

![Image](https://github.com/user-attachments/assets/531e178f-4a7d-4cd9-8f4c-13b30b29aa2a)

**Comment by Cwicbeam** [comment]
Hi there, apologies if this has been answered elsewhere but I can't seem to find it. How do you enable clapper enhancers? I've got both flatpaks installed but nothing has changed in my preferences menu.

**Comment by Rafostar** [comment]
> Hi there, apologies if this has been answered elsewhere but I can't seem to find it. How do you enable clapper enhancers? I've got both flatpaks installed but nothing has changed in my preferences menu.

They aren't in any stable release yet. If you want to try devel version, Clapper Flatpak can be downloaded [here](https://github.com/Rafostar/clapper/actions/workflows/flatpak.yml?query=branch%3Amaster) and enhancers extension built from git main is available [here](https://github.com/Rafostar/clapper-enhancers/actions/workflows/flatpak.yml?query=branch%3Amain).

Once both are installed, they should appear in preferences window -> tweaks -> Enhancers.
